test/Mono.Linker.Tests.Cases/RequiresCapability/RequiresAttributeMismatch.cs

```diff
@@ -0,0 +1,438 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Helpers;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+using Mono.Linker.Tests.Cases.RequiresCapability.Dependencies;
+
+namespace Mono.Linker.Tests.Cases.RequiresCapability
+{
+	[SetupLinkerAction ("copy", "lib1")]
+	[SetupCompileBefore ("lib1.dll", new[] { "Dependencies/ReferenceInterfaces.cs" })]
+	[KeptAllTypesAndMembersInAssembly ("lib1.dll")]
+	[SkipKeptItemsValidation]
+	[ExpectedNoWarnings]
+	class RequiresAttributeMismatch
```

Comment:
I think a better name would be something like `RequiresOnVirtualHierarchy` or something along those lines (or "overrides"). "Attribute mismatch" is confusing - I honestly spent about 2minutes figuring out what it meant.

Reply:
I choose the name based on the name we give to the DiagnosticId. For example, 2046 is `RequiresUnreferencedCodeAttributeMismatch`. The whole file tests that there is a mismatch in attributes between one class and another that was either overridden/implemented. For the warnings that refer to things in VirtualHierarchy there is the file RequiresOnVirtualsAndInterfaces

Reply:
OK - it could be just me... I mean now I will remember forever ;-)